cmd/agent/app/builder.go

```diff
@@ -234,6 +235,7 @@ type ProxyBuilderOptions struct {
 	reporter.Options
 	Logger  *zap.Logger
 	Metrics metrics.Factory
+	Context context.Context
```

Comment:
it's anti-pattern to store context in structs, unless it's explicitly kept as a cancellation signal, which is not the case here